src/dbnode/storage/index/types.go

```diff
@@ -74,6 +84,19 @@ func (o QueryOptions) LimitExceeded(size int) bool {
 	return o.Limit > 0 && size >= o.Limit
 }
 
+// AggregationOptions enables users to specify constraints on aggregations.
+type AggregationOptions struct {
+	QueryOptions
+	TermFilter AggregateTermFilter
+	Type       AggregationType
+}
+
+// LimitExceeded returns whether a given size exceeds the limit
+// the query options imposes, if it is enabled.
+func (o AggregationOptions) LimitExceeded(size int) bool {
```

Comment:
the embedding should give this to you fo free iirc

Reply:
Oh nice :+1: